From e27bfaf49c1483e2498da120398f2e0d2c37fab9 Mon Sep 17 00:00:00 2001
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Wed, 12 Apr 2023 21:39:37 +0100
Subject: [PATCH 4/4] gnu: python-astroalign: Update to 2.4.2.

* gnu/packages/astronomy.scm (python-astroalign): Update to 2.4.2.
  [source]: Use git checkout instead PyPI to enable tests.
  [build-system]: Use pyproject-build-system.
  [propagated-inputs]: Migrate from inputs. Remove labels. Add optional python-bottleneck.
  [native-inputs]: Add python-astropy, python-ccdproc, python-pillow.
---
 gnu/packages/astronomy.scm | 30 +++++++++++++++++-------------
 1 file changed, 17 insertions(+), 13 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index 9961e19695..0d7212791a 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -3070,22 +3070,26 @@ (define-public python-gwcs
 (define-public python-astroalign
   (package
     (name "python-astroalign")
-    (version "2.3.1")
+    (version "2.4.2")
     (source
      (origin
-       (method url-fetch)
-       (uri (pypi-uri "astroalign" version))
+       ;; There are no tests in the PyPI tarball.
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/quatrope/astroalign")
+             (commit (string-append "v" version))))
+       (file-name (git-file-name name version))
        (sha256
-        (base32 "19qzv3552lgrd9qmj0rxs51wmx485hw04cbf76ds5pin85kfaiy1"))))
-    (build-system python-build-system)
-    (arguments
-     ;; TODO: (Sharlatan-20210213T162940+0000): I could not make tests run
-     `(#:tests? #f))
-    (inputs
-     `(("numpy" ,python-numpy)
-       ("scikit-image" ,python-scikit-image)
-       ("scipy" ,python-scipy)
-       ("sep" ,python-sep)))
+        (base32 "0hly20a65540hr3l1lsd1i4d90a0vdrbwnn6zx3z8s89ha9lq3pb"))))
+    (build-system pyproject-build-system)
+    (native-inputs
+     (list python-astropy python-ccdproc python-pillow))
+    (propagated-inputs
+     (list python-bottleneck
+           python-numpy
+           python-scikit-image
+           python-scipy
+           python-sep))
     (home-page "https://astroalign.readthedocs.io/")
     (synopsis "Astrometric Alignment of Images")
     (description
-- 
2.39.2

